src/utl/src/ScopedTemporaryFile.cpp

```diff
@@ -45,15 +45,13 @@ StreamHandler::StreamHandler(const char* filename, bool binary)
   }
   os_.exceptions(std::ofstream::failbit | std::ofstream::badbit);
   try {
-    if (binary) {
-      os_.open(tmp_filename, std::ios::binary | std::ios_base::out);
-    } else {
-      os_.open(tmp_filename, std::ios_base::out);
-    }
+    os_.open(tmp_filename,
+             (binary ? (std::ios::binary | std::ios_base::out)
+                     : std::ios_base::out));
   } catch (std::ios_base::failure& e) {
     std::string error = e.what();
-    throw std::ios_base::failure(error + " (failed to open '" + tmp_filename
-                                 + "')");
+    std::throw_with_nested(std::ios_base::failure(error + " (failed to open '"
+                                                  + tmp_filename + "')"));
```

Comment:
did you try it? What does the stack trace now look like?

Reply:
- Please see stack trace for both cases below:

throw:
```
#0  0x00007ffff78969fc in pthread_kill () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x00007ffff7842476 in raise () from /lib/x86_64-linux-gnu/libc.so.6
#2  0x00007ffff78287f3 in abort () from /lib/x86_64-linux-gnu/libc.so.6
#3  0x00007ffff7ca2b9e in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00007ffff7cae20c in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x00007ffff7cae277 in std::terminate() () from /lib/x86_64-linux-gnu/libstdc++.so.6
#6  0x00007ffff7cae4d8 in __cxa_throw () from /lib/x86_64-linux-gnu/libstdc++.so.6
#7  0x00005555555568c9 in StreamHandler::StreamHandler(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool) ()
#8  0x0000555555556be6 in main ()
```
std::throw_with_nested:
```
#0  0x00007ffff78969fc in pthread_kill () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x00007ffff7842476 in raise () from /lib/x86_64-linux-gnu/libc.so.6
#2  0x00007ffff78287f3 in abort () from /lib/x86_64-linux-gnu/libc.so.6
#3  0x00007ffff7ca2b9e in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00007ffff7cae20c in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x00007ffff7cae277 in std::terminate() () from /lib/x86_64-linux-gnu/libstdc++.so.6
#6  0x00007ffff7cae4d8 in __cxa_throw () from /lib/x86_64-linux-gnu/libstdc++.so.6
#7  0x00005555555574d0 in void std::__throw_with_nested_impl<std::ios_base::failure[abi:cxx11]>(std::ios_base::failure[abi:cxx11]&&, std::integral_constant<bool, true>) ()
#8  0x000055555555723c in void std::throw_with_nested<std::ios_base::failure[abi:cxx11]>(std::ios_base::failure[abi:cxx11]&&) ()
#9  0x0000555555556930 in StreamHandler::StreamHandler(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool) ()
#10 0x0000555555556c4a in main ()
```

- Please see the error messages for both cases below:

throw:
```
Exception: basic_ios::clear: iostream error (failed to open '/invalid/path/example.txt.tmp'): iostream error
```
std::throw_with_nested:
```
Exception: basic_ios::clear: iostream error (failed to open '/invalid/path/example.txt.tmp'): iostream error
 Exception: basic_ios::clear: iostream error
```

Reply:
👍 The exception handler in OpenROAD needs to be updated, but I think that is out of scope for this PR.